homeassistant/components/demo/notify.py

```diff
@@ -1,27 +1,53 @@
 """Demo notification service."""
-from homeassistant.components.notify import BaseNotificationService
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.notify import BaseNotificationService, NotifyService
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.service_platform import AddServicesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
 EVENT_NOTIFY = "notify"
 
 
-def get_service(hass, config, discovery_info=None):
-    """Get the demo notification service."""
-    return DemoNotificationService(hass)
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_services: AddServicesCallback,
+) -> None:
+    """Set up the Demo config entry."""
+    async_add_services([DemoNotifyService()])
 
 
-class DemoNotificationService(BaseNotificationService):
-    """Implement demo notification service."""
+def get_service(
+    hass: HomeAssistant,
+    config: ConfigType,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> BaseNotificationService:
+    """Get the legacy demo notification service."""
+    return DemoNotificationService()
+
+
+class DemoNotifyService(NotifyService):
```

Comment:
Let's make these names more distinguishable. Maybe add word Legacy to the other one.